### DISCUSSION TITLE
Change Chance of Learning Secondary Skills

### DISCUSSION THREAD

**Discussion by PeterLyall5** [ihhub/fheroes2#9400] [General]
Does anyone know how, or if there is the possibilty, to change the chances of learning some secondary skills? I get that it may be a personal preference, but I have had countless games ruined by given the choice between bad secondary skills like Eagle Eye or Mysticism. For example, I like building up Wizards but the Archery skill has one of the lowest chances of being offered to Wizards but Titans are one of only two boss creatures in the entire game and the damage done during battle would be increased greatly if the hero has Archery. I have had so many games where the secondary skills are being perfectly built up and then the hero gets ruined by the choice being offered.

Skills I like:

Wisdom
Logistics
Pathfinding
Luck
Leadership
Archery
Ballistics
Estates

Skills I don't like:

Diplomacy
Eagle Eye
Mysticism
Scouting

Any advice or remedy for changing the chances for some secondary skills?

**Comment by oleg-derevenetz** [comment]
> Any advice or remedy for changing the chances for some secondary skills?

You can change their weights by editing the `src/fheroes2/game/game_static.cpp`. Please note the `factionProperties` - the structure of its members is defined in the file `src/fheroes2/heroes/skill_static.h`. You will need to edit the `weightsOfSecondarySkills` field of the corresponding faction and then recompile the engine.

> For example, I like building up Wizards but the Archery skill has one of the lowest chances of being offered to Wizards but Titans are one of only two boss creatures in the entire game and the damage done during battle would be increased greatly if the hero has Archery.

It would not be very fair from the game balance point of view to have both strong magic, which is taught to wizards, and combat skills (like Archery) at the same time. Of course, you can change anything for yourself, but these changes will not go to the `master` branch.

**Reply by PeterLyall5** [reply]
Thank you for your comments. I’m not a programmer at all, but will definitely try what you said.

**Reply by PeterLyall5** [reply]
Hi Oleg, where is the src/fheroes2/game/game_static.cpp that I can edit the chance weights? I’m not finding it anywhere. Sorry, I’m not a programmer at all. Thanks.

**Reply by oleg-derevenetz** [reply]
> where is the src/fheroes2/game/game_static.cpp that I can edit the chance weights? I’m not finding it anywhere.

Here it is:

https://github.com/ihhub/fheroes2/blob/master/src/fheroes2/game/game_static.cpp

You should download the fheroes2 sources, edit this file and then build fheroes2 binary from these sources.

**Reply by PeterLyall5** [reply]
Hi Oleg, I just wanted to say thanks for all your advice and guidance. I did exactly as you said and was able to edit the game.static.cpp file, save my changes and rebuild the game. I tested it out and it worked. Thanks again.
